llama_stack/core/stack.py

```diff
@@ -321,6 +321,10 @@ def add_internal_implementations(impls: dict[Api, Any], run_config: StackRunConf
     )
     impls[Api.conversations] = conversations_impl
 
+    # Set prompts API on agents provider if it exists
+    if Api.agents in impls and hasattr(impls[Api.agents], "set_prompts_api"):
```

Comment:
the way to do this is to declare the dependency in https://github.com/llamastack/llama-stack/blob/main/llama_stack/providers/registry/agents.py#L31